citations-api/api/src/java/org/sakaiproject/citation/api/CitationCollectionOrder.java

```diff
@@ -0,0 +1,110 @@
+package org.sakaiproject.citation.api;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Created by nickwilson on 6/23/15.
+ */
+public class CitationCollectionOrder {
+
+	private String collectionId;
+	private String citationid;
+	private int location;
+	private SectionType sectiontype;
+	private String value;
+	private List<CitationCollectionOrder> children = new ArrayList<CitationCollectionOrder>();
+
+	public enum SectionType {
+		HEADING1, HEADING2, HEADING3, DESCRIPTION, CITATION
+	}
+
+	public CitationCollectionOrder() {
+	}
+
+	public CitationCollectionOrder(String collectionId, String citationid, int location, SectionType sectiontype, String value) {
+		this.collectionId = collectionId;
+		this.citationid = citationid;
+		this.location = location;
+		this.sectiontype = sectiontype;
+		this.value = value;
+	}
+
+	public CitationCollectionOrder(String collectionId, int location, SectionType sectiontype, String value) {
+		this.collectionId = collectionId;
+		this.location = location;
+		this.sectiontype = sectiontype;
+		this.value = value;
+	}
+
+	public List<CitationCollectionOrder> getChildren() {
+		return children;
+	}
+
+	public void setChildren(List<CitationCollectionOrder> children) {
+		this.children = children;
+	}
+
+	public void addChild(CitationCollectionOrder citationCollectionOrder) {
+		this.children.add(citationCollectionOrder);
+	}
+
+	public String getCitationid() {
+		return citationid;
+	}
+
+	public void setCitationid(String citationid) {
+		this.citationid = citationid;
+	}
+
+	public SectionType getSectiontype() {
+		return sectiontype;
+	}
+
+	public void setSectiontype(SectionType sectiontype) {
+		this.sectiontype = sectiontype;
+	}
+
+	public String getCollectionId() {
+		return collectionId;
+	}
+
+	public void setCollectionId(String collectionId) {
+		this.collectionId = collectionId;
+	}
+
+	public int getLocation() {
+		return location;
+	}
+
+	public void setLocation(int location) {
+		this.location = location;
+	}
+
+	public String getValue() {
+		return value;
+	}
+
+	public void setValue(String value) {
+		this.value = value;
+	}
+
+	public List<CitationCollectionOrder> flatten() {
```

Comment:
Yes you're right.  Really it's kind of two different 'views' of the db.  A list view and a tree view.  I'm using the list view to do "inserts at a location" which is far easier than navigating through a tree and changing the tree's structure - though it could be done this way.  I'm using the tree view for operations like delete - as when you delete an element you also need to delete all of its children (which you can't tell from a list view).